src/hotspot/share/cds/filemap.cpp

```diff
@@ -2047,6 +2047,16 @@ void FileMapInfo::fixup_mapped_heap_regions() {
     assert(open_archive_heap_ranges != NULL, "NULL open_archive_heap_ranges array with non-zero count");
     G1CollectedHeap::heap()->fill_archive_regions(open_archive_heap_ranges,
                                                   num_open_archive_heap_ranges);
+
+    // Populate the open archive regions' G1BlockOffsetTableParts. That ensures
+    // fast G1BlockOffsetTablePart::block_start operations for any given address
+    // within the open archive regions when trying to find start of an object
+    // (e.g. during card table scanning).
+    //
+    // This is only needed for open archive regions, but not the closed archive
+    // regions, because objects in closed archive regions are 'immutable'.
```

Comment:
```suggestion
    // This is only needed for open archive regions, but not the closed archive
    // regions, because objects in closed archive regions never reference objects
    // outside the closed archive regions and they are immutable, so we never
    // need their BOT during garbage collection.
```